app/src/main/java/com/google/samples/apps/nowinandroid/ui/AdaptiveScaffold.kt

```diff
@@ -0,0 +1,216 @@
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> AdaptiveScaffold(
+    modifier: Modifier = Modifier,
+    navigationItems: List<T>,
+    navigationItemTitle: @Composable (item: T, isSelected: Boolean) -> Unit,
+    navigationItemIcon: @Composable (item: T, isSelected: Boolean) -> Unit,
+    isItemSelected: @Composable (item: T) -> Boolean,
+    onNavigationItemClick: (item: T) -> Unit,
+    topBar: @Composable () -> Unit = {},
+    snackbarHost: @Composable () -> Unit = {},
+    colors: AdaptiveScaffoldNavigationComponentColors = AdaptiveScaffoldNavigationComponentDefaults.colors(),
+    contentWindowInsets: WindowInsets = ScaffoldDefaults.contentWindowInsets,
+    content: @Composable (padding: PaddingValues) -> Unit,
+) {
+    val context = LocalContext.current
+    val metrics = WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(context)
+    val widthDp = metrics.bounds.width() / context.resources.displayMetrics.density
+
+    when {
+        widthDp >= 1240f -> {
+            Scaffold(
+                modifier = modifier,
+                topBar = topBar,
+                snackbarHost = snackbarHost,
+                containerColor = colors.contentContainerColor,
+                contentColor = colors.contentColor,
+                contentWindowInsets = contentWindowInsets,
+            ) { padding ->
+                PermanentNavigationDrawer(
+                    drawerContent = {
+                        PermanentDrawerSheet(
+                            drawerContainerColor = colors.drawerContainerColor,
+                        ) {
+                            navigationItems.forEach { item ->
+                                NavigationDrawerItem(
+                                    label = { navigationItemTitle(item, isItemSelected(item)) },
+                                    icon = { navigationItemIcon(item, isItemSelected(item)) },
+                                    selected = isItemSelected(item),
+                                    onClick = { onNavigationItemClick(item) },
+                                    colors = NavigationDrawerItemDefaults.colors(
+                                        selectedContainerColor = colors.selectedContainerColor,
+                                        unselectedContainerColor = colors.unselectedContainerColor,
+                                        selectedIconColor = colors.selectedIconColor,
+                                        unselectedIconColor = colors.unselectedIconColor,
+                                        selectedTextColor = colors.selectedTextColor,
+                                        unselectedTextColor = colors.unselectedTextColor,
+                                    ),
+                                )
+                            }
+
+                        }
+                    },
+                    modifier = Modifier.padding(padding),
+                ) {
+                    content(padding)
```

Comment:
Because we're using different instances of `Scaffold` depending on the configuration, that means we'll also lose app state when the window size changes. 

To avoid that, should we wrapped the content in `movableContentOf`? I don't think we have a problem with `NavigationDrawerItem` and other components because they're stateless